File: FrontendRazorPage/Models/VocabularyModel.cs
using Microsoft.AspNetCore.Mvc.ModelBinding.Validation;
namespace FrontendRazorPage.Models
{
    public class VocabularyModel
    {
        public int Id { get; set; }
        public int LessonId { get; set; }
        public string? Kanji { get; set; }
        public string Hiragana { get; set; }
        public string Romaji { get; set; }
        public string Meaning { get; set; }
        public string ExampleSentence { get; set; }
        [ValidateNever]
        public string ? TextToSpeak { get; set; }
    }
}

File: FrontendAngular/src/app/Models/vocabulary-model.ts
export interface VocabularyModel {
    id: number;
    lessonId: number;
    kanji?: string;           // Dấu ? tương đương với string? trong C#
    hiragana: string;
    romaji: string;
    meaning: string;
    exampleSentence?: string; // Dấu ? vì có thể từ vựng không có câu ví dụ
    textToSpeak?: string;
}